tee-worker/litentry/core/assertion-build/src/a4.rs

```diff
@@ -76,73 +73,87 @@ pub fn build(
 				_ => {},
 			}
 		}
+
 		if matches!(
 			verified_network,
 			VerifiedCredentialsNetwork::Litentry
 				| VerifiedCredentialsNetwork::Litmus
 				| VerifiedCredentialsNetwork::LitentryRococo
 				| VerifiedCredentialsNetwork::Ethereum
 		) {
-			let q_min_balance: f64 = if verified_network == VerifiedCredentialsNetwork::Litentry
-				|| verified_network == VerifiedCredentialsNetwork::Litmus
-				|| verified_network == VerifiedCredentialsNetwork::LitentryRococo
-			{
-				(min_balance / (10 ^ 12)) as f64
-			} else {
-				(min_balance / (10 ^ 18)) as f64
-			};
-
-			let mut addresses: Vec<String> = vec![];
-			match &identity {
+			let address = match &identity {
 				Identity::Evm { address, .. } => {
 					let mut address = account_id_to_string(address.as_ref());
 					address.insert_str(0, "0x");
-					debug!("	[AssertionBuild] A4 EVM address : {}", address);
-
-					addresses.push(address);
+					Ok(address)
 				},
 				Identity::Substrate { address, .. } => {
 					let mut address = account_id_to_string(address.as_ref());
 					address.insert_str(0, "0x");
-					debug!("	[AssertionBuild] A4 Substrate address : {}", address);
-
-					addresses.push(address);
+					Ok(address)
 				},
 				Identity::Web2 { address, .. } => match from_utf8(address.as_ref()) {
-					Ok(addr) => addresses.push(addr.to_string()),
-					Err(e) => error!(
-						"	[AssertionBuild] A4 parse error Web2 address {:?}, {:?}",
-						address, e
-					),
+					Ok(addr) => Ok(addr.to_string()),
+					Err(e) => {
+						error!("Assertion A4 parse Web2 address {:?} error info: {:?}", address, e);
+
+						Err(())
+					},
 				},
-			}
-			let mut tmp_token_addr = String::from("");
-			if verified_network == VerifiedCredentialsNetwork::Ethereum {
-				tmp_token_addr = LIT_TOKEN_ADDRESS.to_string();
-			}
+			};
 
-			for (index, from_date) in ASSERTION_FROM_DATE.iter().enumerate() {
-				// if found is true, no need to check it continually
-				if found {
-					from_date_index = index + 1;
-					break
+			if let Ok(address) = address {
+				if let Some(set) = networks.get_mut(&verified_network) {
+					set.insert(address);
 				}
```

Comment:
The map will be forever empty if I'm not mistaken because the initial `get_mut` will always return `None` and it can never insert any element.

Reply:
yes, it's a bug, fixed